### PR TITLE
Add missing `rpmbuild` dependencies

### PR DESCRIPTION
**Pull request by bossmc**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
`rpmbuild` will call out to `patch`, `file` and `diff` (when applying patches, stripping debug symbols and validating codebase cleanliness respectively).  This change makes `rpmbuild` work after a `tdnf install` (or on a `base` image where it's pre-installed, though that feels questionable).

###### Change Log  
- None probably - this is a pretty minimal internal change

###### Does this affect the toolchain?  
No

###### Associated issues  
None

###### Links to CVEs  
None

###### Test Methodology
Local build and install

### CODE REVIEW

**Review by oliviacrain — DISMISSED** [review]
Since `rpm` is in our toolchain, we'll need to update the toolchain manifests - if you're not familiar with the process, I can provide you with a suitable patch when I have some free time this week.

I'll also loop in our colleague who is working on image size reduction, just to double check that there aren't any outsized effects on base image size.

**Comment by bossmc** [comment]
I've now subsumed https://github.com/microsoft/CBL-Mariner/pull/2493/files into this MR - by removing the `rpm-build` dependency from `rpm` these additional packages won't be pulled into the core image (thus net saving size).

File: SPECS/rpm/rpm.spec [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Summary:        Package manager
 Name:           rpm
 Version:        4.17.0
-Release:        5%{?dist}
+Release:        6%{?dist}
```
Comment (anphel31) [inline SPECS/rpm/rpm.spec:4]: Can you bump to release 7, since the package is already at release 6 in the main branch

**Review by anphel31 — DISMISSED** [review]
Can you please rebase these changes on the latest version of the rpm spec. We can then run a full build to help validate the changes.

File: SPECS/rpm/rpm.spec [hunk 2]
```diff
@@ -269,6 +276,10 @@ popd
 %{python3_sitelib}/*
 
 %changelog
+* Tue Apr 12 2022 Andy Caldwell <andycaldwell@microsoft.com> - 4.17.0-7
```
Comment (anphel31) [inline SPECS/rpm/rpm.spec:279]: Can you update the date here so everything's in chronological order

Comment (anphel31) [reply SPECS/rpm/rpm.spec:279]: Hopefully fixing this will also clear up the manifest check issues

File: SPECS/rpm/rpm.spec [hunk 3]
```diff
@@ -269,6 +276,10 @@ popd
 %{python3_sitelib}/*
 
 %changelog
+* Fri May 13 2022 Andy Caldwell <andycaldwell@microsoft.com> - 4.17.0-7
+- Add missing dependencies to rpmbuild (file, diff and patch)
+- Remove spurious `rpm` dependency on `rpm-build`
```
Comment (PawelWMS) [inline SPECS/rpm/rpm.spec:281]: Nit: line probably no longer needed after you rebased on top of your previous changes removing this dependency.

```suggestion
```

**Review by PawelWMS — COMMENTED** [review]
Have someone already helped you with running a full pipeline build? If not, let me know and I can do that for you.

**Comment by bossmc** [comment]
@anphel31 has just run a full pipeline build for me on this change, so I hope this is good to go?

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**